### PR TITLE
library: Port HTTP Request demo to Vala

### PR DESCRIPTION
**Pull request by Diego-Ivan**
Ports the HTTP Request demo to Vala

### CODE REVIEW

File: src/Library/demos/HTTP Request/main.vala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  var date = new DateTime.now_local ();
+  var http_session = new Soup.Session ();
+
+  // https://api.wikimedia.org/wiki/Feed_API/Reference/Featured_content
+  string language = "en";
+  string url =
+    @"https://api.wikimedia.org/feed/v1/wikipedia/$language/featured/$(date.format ("%Y/%m/%d"))";
+
+  var message = new Soup.Message ("GET", url);
+  InputStream input_stream = yield http_session.send_async (message, Priority.DEFAULT, null);
```
Comment (lw64) [inline src/Library/demos/HTTP Request/main.vala:31]: I think its way easier to use `send_and_read_async()` than using an `InputStream`. Was there a specific reason to do so?

File: src/Library/demos/HTTP Request/main.vala [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+private void parse_json_response (string json) throws Error {
+  // https://valadoc.org/json-glib-1.0/Json.Parser.html
+  var parser = new Json.Parser ();
+  parser.load_from_data (json);
+
+  // https://valadoc.org/json-glib-1.0/Json.Reader.html
+  Json.Node root_node = parser.get_root ();
+  var reader = new Json.Reader (root_node);
```
Comment (lw64) [inline src/Library/demos/HTTP Request/main.vala:61]: Can you provide a rough structure of the recieved json document in a comment, so the parsing is easier to understand. Also the cursor based API isnt intuitive, so a short explaination how it works would be good I think. (or maybe just accessing it normally with nodes would be easier?)

**Review by lw64 — CHANGES_REQUESTED** [review]

File: src/Library/demos/HTTP Request/main.vala (hunk 1 above)
Comment (Diego-Ivan) [reply src/Library/demos/HTTP Request/main.vala:31]: Just for consistency with the JS demo, as it uses `send_async` too and reads the response using an `InputStream`, but yeah, `send_and_read_async` is way easier and probably what should be done in the first place. I will use `send_and_read_async` and send a PR to use it in the JS demo later.

**Review by lw64 — APPROVED** [review]

File: src/Library/demos/HTTP Request/main.vala (hunk 1 above)
Comment (lw64) [reply src/Library/demos/HTTP Request/main.vala]: Thanks! That is of course the best solution :)

**Pull request merged**